### ISSUE TITLE
Typescript errors

### ISSUE THREAD

**Issue by kokushkin** [styled-icons/styled-icons#1355]
Hey, I'm getting multiple typescript errors, like

```
    ERROR in C:/REPOGIT/webx.partners/node_modules/@styled-icons/icomoon/Copy/Copy.tsx
    ERROR in C:/REPOGIT/webx.partners/node_modules/@styled-icons/icomoon/Copy/Copy.tsx(11,6):
    TS2769: No overload matches this call.
      Overload 1 of 2, '(props: Pick<Pick<Pick<StyledIconProps & StyledIconBaseProps, "string" | "children" | "style" | "title" | "clipPath" | "filter" | "mask" | "path" | ... 465 more ... | "iconVerticalAlign"> & RefAttributes<...>, "string" | ... 473 more ... | "iconVerticalAlign"> & Partial<...>, "string" | ... 473 more ... | "iconVerticalAlign"> & { ...; } & { ...; }): ReactElement<...>', gave the following error.
        Type '{ children: Element; ref: Ref<SVGSVGElement>; size?: string | number | undefined; title?: string | null | undefined; string?: string | number | undefined; ... 417 more ...; iconViewBox: string; }' is missing the following properties from type 'Pick<Pick<Pick<StyledIconProps & StyledIconBaseProps, "string" | "children" | "style" | "title" | "clipPath" | "filter" | "mask" | "path" | ... 465 more ... | "iconVerticalAlign"> & RefAttributes<...>, "string" | ... 473 more ... | "iconVerticalAlign"> & Partial<...>, "string" | ... 473 more ... | "iconVerticalAlign">': path, "aria-activedescendant", "aria-atomic", "aria-autocomplete", and 48 more.
      Overload 2 of 2, '(props: StyledComponentPropsWithAs<ForwardRefExoticComponent<Pick<StyledIconProps & StyledIconBaseProps, "string" | "children" | "style" | "title" | "clipPath" | "filter" | "mask" | "path" | ... 465 more ... | "iconVerticalAlign"> & RefAttributes<...>>, any, {}, never>): ReactElement<...>', gave the following error.
        Type '{ children: Element; ref: Ref<SVGSVGElement>; size?: string | number | undefined; title?: string | null | undefined; string?: string | number | undefined; ... 417 more ...; iconViewBox: string; }' is missing the following properties from type 'Pick<Pick<Pick<StyledIconProps & StyledIconBaseProps, "string" | "children" | "style" | "title" | "clipPath" | "filter" | "mask" | "path" | ... 465 more ... | "iconVerticalAlign"> & RefAttributes<...>, "string" | ... 473 more ... | "iconVerticalAlign"> & Partial<...>, "string" | ... 473 more ... | "iconVerticalAlign">': path, "aria-activedescendant", "aria-atomic", "aria-autocomplete", and 48 more.
```
This how I'm importing the icon:
```
import { Copy } from "@styled-icons/icomoon/Copy/Copy.js";
```
(have to use .js because overvise I'm getting error with es6 when uglifying)

So it looks like I'm trying to compile Copy.tsx! Yeah, take a look:
![image](https://user-images.githubusercontent.com/9853656/89067668-6a3adc00-d378-11ea-81f9-a32cc8830b7f.png)

We have tsx files in the package. Is it ok to have them? 
If I delete these tsx - everything fine!

I'm on  
"styled-components": "4.3.2",
"styled-icons": "10.13.0",

**Comment by jacobwgillespie** [comment]
Hey @kokushkin, thanks for the report - can you share what versions of `@types/react` and `@types/styled-components` you're using? They are likely involved in this as well.

Also would you be able to provide a repo demonstrating the issue? I can do some debugging without it, but if you do have a small example repo to share that would speed things up.

**Comment by kokushkin** [comment]
Thank you for the response and thank you for the library of course!) 
Yeah, sure, it should be even more important, so
"@types/react": "16.8.8",
"@types/styled-components": "4.1.22",

Yeah I will try to reproduce the small example  (unfortunately I can't share the original repo) but it will take some time and it's complicated because we're using quite unique things and I can't use them in the example so I have to mock them somehow.

But, It's a little strange for me why I see tsx files in the library. In fact it's almost the only library  in our repo that contains .tsx files (99.97% of .tsx files in node_module come from your library). And we are using about 1450 libraries.. So could you please comment on why are you keeping the tsx  files in your module?

**Comment by jacobwgillespie** [comment]
Oh, I see now, thanks for the clarification, somehow my brain registered the `.tsx` files earlier but didn't make the connection "those shouldn't be there".

I'll push an updated release without the `.tsx` files 👍

**Comment by jacobwgillespie** [comment]
This has been released as 10.14.1, let me know if you run into any issues.

**Comment by kokushkin** [comment]
Yeah, great, it's worked out. And additionally, it's solved another our problem, so I can include like this
import { Edit } from "@styled-icons/boxicons-regular/Edit/Edit"
instead of doing
import { Edit } from "@styled-icons/boxicons-regular/Edit/Edit.js"
(had to use .js because otherwise I was getting an error with es6 when uglifying)
As I understood I was building tsx and was using the output of this build  instead of your compiled  .js files. But it's just my guessings.
So thank you again! I'll try to contribute somehow.
